lib/src/rules/unnecessary_overrides.dart

```diff
@@ -127,36 +140,55 @@ abstract class _AbstractUnnecessaryOverrideVisitor extends SimpleAstVisitor {
     rule.reportLint(declaration.name);
   }
 
+  /// Returns whether [declaration] is annotated with any metadata (other than
+  /// `@override` or `@Override`).
   bool _addsMetadata() {
     var metadata = declaration.declaredElement?.metadata;
     if (metadata != null) {
       for (var annotation in metadata) {
-        if (!annotation.isOverride) {
-          return true;
-        }
+        if (annotation.isOverride) continue;
+        if (annotation.isProtected && _inheritedMethod.hasProtected) continue;
+        // TODO(srawlins): The above check probably holds for any annotation
+        // which has a private constructor. That is, overriding
```

Comment:
It isn't particularly important, but I think the criteria ought to be "zero arguments" or "identical arguments" instead of "private constructor", though that probably only applies reliably to our current set of annotations. I could see an annotation that was defined in such a way that it was not inherited and hence needed to be repeated at different layers in the hierarchy. Generalizing in this situation is probably error-prone, so this might not need to be a TODO.

Reply:
Good point. I agree that there are subtle bugs that lie this way. The `protected` request comes from google internal, so I know it has an effect, and that people are expanding visibility this way.

Reply:
Yeah, my comment was purely about the TODO, not about the code.

One question: Is `@protected` inherited, and if so does overriding actually stop the inheritance?